syntax = "proto3";

package recommendation;

option go_package = "guru-game/gRPC";

import "google/protobuf/timestamp.proto";

message UserAction {
  string user_id = 1;
  string boardgame_id = 2;
  string action_type = 3;       // เช่น "like", "favorite", "play_time"
  double action_value = 4;      // เช่น เวลาที่เล่น หรือ 1.0 สำหรับ like
  google.protobuf.Timestamp action_time = 5;  // เวลาที่กระทำกิจกรรม
}

message RecommendationRequest {
  string user_id = 1;
  int32 limit = 2;  // จำนวนรายการแนะนำ
}

message Boardgame {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 min_players = 4;
  int32 max_players = 5;
  int32 play_time_min = 6;
  int32 play_time_max = 7;
  string categories = 8;
  double rating_avg = 9;
  int32 rating_count = 10;
  double popularity_score = 11;
  string image_url = 12;
}

message RecommendationResponse {
  repeated Boardgame boardgames = 1;
}

message Response {
  bool success = 1;
  string message = 2;
}

message BoardgamesRequest {
  repeated Boardgame boardgames = 1;
}

service RecommendationService {
  rpc SendUserAction(UserAction) returns (Response);
  rpc GetRecommendations(RecommendationRequest) returns (RecommendationResponse);
  rpc SendAllBoardgames(BoardgamesRequest) returns (Response);
}